source/features/update-pr-from-base-branch.tsx

```diff
@@ -50,13 +50,13 @@ async function addButton(position: Element): Promise<void> {
 	const {base, head} = getBranches();
 	const [pr, comparison] = await Promise.all([
 		getPrInfo(),
-
-		// TODO: Find how to determine whether the branch needs to be updated via v4
-		// `page=10000` avoids fetching any commit information, which is heavy
-		api.v3(`compare/${base}...${head}?page=10000`),
+		// TODO: Drop v3 when GHE supports this via v4
+		pageDetect.isEnterprise()
+			? api.v3(`compare/${base}...${head}?page=10000`) // `page=10000` avoids fetching any commit information, which is heavy
+			: getPrBranchAheadStatus(base),
 	]);
```

Comment:
We should make obsolete code self-contained so that when the time comes we can just delete lines and have a negative diff, without having to delve into the API.

Since this now requires only one call, this piece of code should just be:

```js
const pr = await getPrInfo();
```

In it, we can have the GHE condition that makes two calls if necessary, like:

```js
function getPrInfo() {
	if(GHE) {
		return {
			other: await api0(),
			compare: await api1()
		}
	}

	return api.v4(our new query)
}

In the current code, everyone still gets two requests, so this PR becomes of little utility for GitHub.com users at this point.

Reply:
> In the current code, everyone still gets two requests, so this PR becomes of little utility for GitHub.com users at this point.

The problem is the I want to use this utility for `restore-file` but if I combine the requests then I will not be able to use it.

Your call

Reply:
It would be nice but not worth it. Your reversal looks good